---
title: ""
author: ""
date: ""
output:
  html_document:
    toc: true
    toc_float: true
    collapsed: false
    theme: cerulean
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(dplyr)
library(ggpubr)
library(gridExtra)
library(grid)
library(cowplot)
library(reactable)
library(data.table)
library(reticulate)
library(sciplot)
library(plotly)
library(matrixStats)
library(SummarizedExperiment)
library(RColorBrewer)
library(heatmaply)
library(ggrepel)
library(tidyverse)
library(gprofiler2)
library(patchwork)
library(DESeq2)
library(apeglm)
library(fgsea)
library(msigdbr)
library(compiler)
library(shiny)
library(tippy)
library(msigdbr)
library(GSVA)
library(pals)
library(ggtext)
library(broom)
library(DT)
library(EnsDb.Hsapiens.v79)
library(org.Hs.eg.db)
library(eulerr)
library(GeneOverlap)
#load RData
load("~/Documents/Q005841_HNSCC/work/processed_dat/dds_Dec2021_hnscc.RData")
load("~/Documents/Q005841_HNSCC/work/processed_dat/rld_hnscc_Dec2021.RData")
```
# Cal27: NSD2.KO vs NSD1.KO {.tabset .tabset-pills}

<b> NSD1.KO is set as baseline/reference. Cutoff for DEGs: abs(FC) > 2 & padj < 0.05 </b>

We find more upregulated genes in NSD2.KO than downregulated ones. However, overrepresentation of similar pathways are found in upregulated and downregulated genes, such as cell periphery and multicellular organismal process. However, downregulated genes are also enriched in pathways in cancer & calcium signaling.

```{r include= FALSE}
get_dge_df <- cmpfun(function(dds,group,sampA,ref_sampB){ #group is similar to condition
  df <- dds %>%
    results(contrast = c(group,sampA,ref_sampB)) %>%
    data.frame() %>%
    na.omit() %>%
    rownames_to_column('ensembl_gene_id')
  geneSym <- ensembldb::select(EnsDb.Hsapiens.v79, keys=df$ensembl_gene_id, keytype = "GENEID", columns = c("SYMBOL","GENEID")) #this gets human gene symbols for ensembl gene ids
  colnames(geneSym) <- c("gene","ensembl_gene_id")
  df_final <- full_join(df,geneSym,by="ensembl_gene_id")
  return(df_final)
})
#
volc <- function(r, x, y, ylab, ttl) {
  d <- as.data.frame(r) %>%
    dplyr::rename(x = !!x, y = !!y) %>%
    mutate(kind = case_when((abs(x) > 2 & y < .05) ~ 'DE',
                            abs(x) > 2 ~ '|logFC|>2',
                            y < .05 ~ 'FDR<0.05',
                            T ~ 'NS'),
           y = -log10(y))
  ct <- d %>% 
    dplyr::filter(kind == 'DE') %>%
    mutate(up = x > 0) %>%
    dplyr::count(up) %>%
    mutate(x = ifelse(up, Inf, -Inf),
           y = Inf,
           h = as.numeric(up))
  ggplot(d, aes(x, y, color = kind)) +
    geom_vline(xintercept = c(-2, 2), linetype = 'dashed') +
    geom_hline(yintercept = -log10(0.05), linetype = 'dashed') +
    geom_point(alpha = .5) +
    geom_label(aes(x = x, y = y, label = n, hjust = h),
              vjust = 1, data = ct, inherit.aes = F) +
     geom_text_repel(aes(label = gene), data = subset(d, kind == "DE"),max.overlaps = 10,show.legend = F, min.segment.length = 0) + scale_y_continuous() +
    scale_color_manual(values = c('orange','forestgreen', 'red',  'black')) +
    labs(x = x, y = ylab, title = ttl) + theme(axis.line = element_line(colour = "black"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        legend.title = element_blank())
}
###overlap analysis###
overlap_analysis <- cmpfun(function(dds,grouping,cond1,cond2,baseCond,fc_cutoff){ #x is dge df; bg = background list
  x1 <- get_dge_df(dds,grouping,cond1,baseCond) 
  x2 <- get_dge_df(dds,grouping,cond2,baseCond)
  bg <- rbind(x1,x2)
  bg <- bg$ensembl_gene_id
  bg <- bg[!duplicated(bg)]
  go.obj <- newGeneOverlap(x1$ensembl_gene_id[x1$padj < .05 & abs(x1$log2FoldChange) > fc_cutoff],x2$ensembl_gene_id[x2$padj < .05 & abs(x2$log2FoldChange) > fc_cutoff],genome.size=length(bg),spec="hg19.gene")
go.obj <- testGeneOverlap(go.obj)
  pa <- gost(go.obj@intersection,
             organism = 'hsapiens',
             custom_bg = bg,
             user_threshold = 0.05,
             significant = TRUE,
             evcodes = TRUE,
             exclude_iea = TRUE,
             correction_method = "fdr",
             sources = c("GO","KEGG"))
  reactable <- pa$result %>%
    dplyr::select(source, term_name, intersection_size, p_value) %>% 
  arrange(p_value) %>%
  setNames(c("Source","Pathway","Number of Genes","FDR"))
  return(list(x1=x1,x2=x2,go.obj=go.obj,pa=pa,reactable=reactable))
})
###eulergraph for 3 groups###
get_euler_graph <- function(ttl1,ttl2,ttl3,ttl4,x1,x2,x3,x4,logfc_cutoff){
x <- list(c1 = x1$ensembl_gene_id[x1$padj < .05 & abs(x1$log2FoldChange) > logfc_cutoff],c2 = x2$ensembl_gene_id[x2$padj < .05 & abs(x2$log2FoldChange) > logfc_cutoff],c3 = x3$ensembl_gene_id[x3$padj < .05 & abs(x3$log2FoldChange) > logfc_cutoff],c4 = x4$ensembl_gene_id[x4$padj < .05 & abs(x4$log2FoldChange) > logfc_cutoff])
names(x) <- c(ttl1,ttl2,ttl3,ttl4)
p <- plot(euler(x),legend=FALSE,quantities=TRUE)
return(p)
}
###eulergraph for 2 groups####
get_euler_graph_for_two <- function(ttl1,ttl2,x1,x2,logfc_cutoff){
x <- list(c1 = x1$ensembl_gene_id[x1$padj < .05 & abs(x1$log2FoldChange) > logfc_cutoff],c2 = x2$ensembl_gene_id[x2$padj < .05 & abs(x2$log2FoldChange) > logfc_cutoff])
names(x) <- c(ttl1,ttl2)
p <- plot(euler(x),legend=FALSE,quantities=TRUE)
return(p)
}
```

## Volcano
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
#How does NSD2-KO differ from NSD1-KO?
#resLFC_cal27_nsd1v2 <- lfcShrink(dds, contrast = c('group', 'Cal27NSD2.KO', 'Cal27NSD1.KO'), type = 'normal')
load("~/Documents/Q005841_HNSCC/data/rd_cal27_nsd1v2.RData")
#volcano plot
volc(rd_cal27,'log2FoldChange','padj','-log10(FDR)','')
```

## Pathway analysis - upregulated
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
#pathway analysis
load("~/Documents/Q005841_HNSCC/data/cal27_pa.RData")

gostplot(cal27_pa$up,capped = FALSE,interactive = TRUE)
reactable(cal27_pa$up_reactable)
```

## Pathway analysis - downregulated
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
gostplot(cal27_pa$down,capped = FALSE,interactive = TRUE)
reactable(cal27_pa$down_reactable)
```

## GSEA

Downregulated genes are enriched in oxidative phosphorylation and p53 pathway. Upregulated genes are enriched in epithelial mesechymal transition. 

```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
load("~/Documents/Q005841_HNSCC/data/cal27_nsd1v2_gsea.RData")
plot(cal27_nsd1v2_gsea$plot.r)
cal27_nsd1v2_gsea$gsea_table
```

## GSEA top hit
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
plot(cal27_nsd1v2_gsea$top_hit)
```

## {-}

---

<p>&nbsp;</p>

# Cal27: comparison of NSD knockouts {.tabset .tabset-pills}

## Volcano

<b> DEG cutoff for the volcano plots was FDR < 0.05 & |log2FC| > 2 </b>

NSD.DKO has few DEGs when compared to NSD1KO. Largest number of DEGs found between TKO and NSD2KO. 

```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
cs <- c('Cal27NSD1.KO', 'Cal27NSD2.KO', 'Cal27NSD.DKO', 'Cal27NSD.TKO')
sclrs <- c(DE = 'forestgreen', 'FDR<0.05' = 'red', '|logFC|>2' = 'orange', 'NS' = 'black')
load("~/Documents/Q005841_HNSCC/data/nsd_comparisons.RData")
load("~/Documents/Q005841_HNSCC/data/colorsfor_nsdcomparisons.RData")
ggplot(pd, aes(x = log2FoldChange, y = -log10(padj))) +
  geom_point(aes(color = kind), size = 0.1, alpha = 0.7) +
  scale_color_manual(values = sclrs) +
  geom_richtext(aes(label = l), data = sts, x = 0, y = 35,size = 3) +
  facet_grid(c2 ~ c1) +
  xlab(expression(log[2]*FoldChange)) +
  ylab(expression(-log[10](p[adj]))) +
  coord_cartesian(ylim = c(0,60)) + geom_text_repel(aes(label = gene), data = subset(pd, kind == "DE"),max.overlaps = 5,show.legend = F, min.segment.length = 0,force = 0.1,force_pull = 0.1,size=1.5) +
  theme(legend.position = 'none',
        plot.background = element_blank(),
        panel.background = element_rect(fill = NA, color = 'black', size = 1),
        axis.text = element_text(color = 'black'),
        panel.grid = element_blank(),
        #axis.line = element_line(color = 'black'),
        panel.grid.major = element_line(color = 'grey90', linetype = 'dashed'),
        strip.background = element_rect(fill='black'),
        strip.text = element_text(color = 'white'),
        legend.title = element_blank(),
        legend.background = element_blank(),
        legend.key = element_blank())
```

## Overlap {.tabset}

We evaluate how these DEGs behave in different contrast overlap. The criteria for DEG here is only padj < 0.05.

Essentially we recapitulate what we see from PCA and volcano plots, in that NSD1KO is not so different from NSD.DKO. 

### Odds ratio 
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
load("~/Documents/Q005841_HNSCC/data/res_nsd_ko_comparisons.RData")
nums <- res %>%
  na.omit() %>%
  dplyr::filter(padj < .05) %>%
  mutate(s = case_when(log2FoldChange > 0 ~ sprintf('%s > %s', c1, c2),
                       log2FoldChange < 0 ~ sprintf('%s < %s', c1, c2)),
         d = case_when(log2FoldChange > 0 ~ '+',
                       log2FoldChange < 0 ~ '-')) %>%
  {split(.$ID, .$s)}
ugs <- unlist(nums) %>%
  unique() %>%
  length()
ors <- lapply(names(nums), function(x) {
  lapply(names(nums), function(y) {
    mm <- tibble(a = length(intersect(nums[[x]],nums[[y]]))) %>%
      mutate(b = length(nums[[x]]) - a,
             c = length(nums[[y]]) - a,
             d = ugs - a - b - c)
      
    matrix(unlist(mm), 2) %>%
      {mutate(tidy(fisher.test(., alternative = 'two.sided')),
              p = fisher.test(., alternative = 'greater')$p.value)} %>%
      dplyr::select(or = estimate, pb = p.value,
                    clo = conf.low, chi = conf.high, p) %>%
      mutate(c1 = x, c2 = y) %>%
      cbind(mm)
  }) %>% bind_rows()
}) %>% bind_rows() %>% as_tibble() %>% mutate(j = a / (a + b + c))
hms <- c('p','or','j') %>%
  setNames(., .) %>%
  lapply(function(x) {
    oo <- ors[,c('c1','c2',x)] %>%
      `colnames<-`(c('c1','c2','v'))
    if (x == 'p') {
      oo$v <- oo$v %>%
        {.[.==0] <- min(.[.!=0]);.} %>%
        log10() %>%
        {-.}
      oo %>%
        mutate(v = case_when(!is.finite(v)~NA_real_, T ~ v)) %>%
        pivot_wider(names_from = c2, values_from = 'v') %>%
        column_to_rownames('c1') %>%
        heatmaply()
    } else if (x == 'or') {
     
      oo$v <- log2(oo$v)
      oo$v[oo$v == -Inf] <- min(oo$v[is.finite(oo$v)])
      oo$v[oo$v == Inf] <- max(oo$v[is.finite(oo$v)])
      oo %>%
        mutate(v = case_when(!is.finite(v)~NA_real_, T ~ v)) %>%
        pivot_wider(names_from = c2, values_from = 'v') %>%
        column_to_rownames('c1') %>%
        heatmaply(scale_fill_gradient_fun = scale_fill_gradientn(
          colors = pals::coolwarm(10), limits = c(-5,5)))
    } else {
      oo %>%
        pivot_wider(names_from = c2, values_from = 'v') %>%
        column_to_rownames('c1') %>%
        heatmaply()
    }
  })
hms$or
```

### Fisher p-value
```{r echo=FALSE, message=FALSE, warning=FALSE,out.width='100%',fig.height=6}
hms$p
```

### Jaccard indices
```{r echo=FALSE, message=FALSE, warning=FALSE,out.width='100%',fig.height=6}
hms$j
```

### {-}

## Pathway Analysis  {.tabset}

With these 4 conditions we can also ask if the genes consistently up- or down-regulated in a given condition relative to all others are enriched in certain pathways. Cutoff criteria FDR < 0.05 and |log2FC| > 1

NSD1.KO and NSD.DKO had few DEGs relative to all others and few overrepresented pathways. On the other hand, NSD2.KO and NSD.TKO had a greater number of DEGs and a few overrepresented pathways. 

### Cal27NSD1.KO {.tabset} 
```{r echo=FALSE, message=FALSE, warning=FALSE}
load("~/Documents/Q005841_HNSCC/data/ug_cal27_nsd.RData")
load("~/Documents/Q005841_HNSCC/data/uu_cal27_nsd.RData")
load("~/Documents/Q005841_HNSCC/data/rr_cal27_nsd.RData")
cnd <- 'Cal27NSD1.KO'
```

#### Higher than others 

There were `r length(ug[[paste0(cnd, '.up')]])` such genes, but no enriched pathways.
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
#gostplot(rr[[paste0(cnd, '.up')]])
```

#### Lower than others

There were `r length(ug[[paste0(cnd, '.dn')]])` such genes

```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
gostplot(rr[[paste0(cnd, '.dn')]])
```

```{r echo=FALSE,out.width='100%'}
rr[[paste0(cnd, '.dn')]]$result %>% 
           dplyr::select(term_name, source, term_size, query_size, 
                         intersect_size = intersection_size, p = p_value) %>% 
           mutate(source = factor(source)) %>%
           datatable(style = 'bootstrap', class = 'table-hover table-condensed', 
                     rownames = FALSE, filter = 'top') %>%
           formatRound(columns = c('p'), digits=3)
```

### Cal27NSD2.KO {.tabset}
```{r echo=FALSE, message=FALSE, warning=FALSE}
cnd <- 'Cal27NSD2.KO'
```

#### Higher than others

There were `r length(ug[[paste0(cnd, '.up')]])` such genes

```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
gostplot(rr[[paste0(cnd, '.up')]])
```

```{r echo=FALSE,out.width='100%'}
rr[[paste0(cnd, '.up')]]$result %>% 
           dplyr::select(term_name, source, term_size, query_size, 
                         intersect_size = intersection_size, p = p_value) %>% 
           mutate(source = factor(source)) %>%
           datatable(style = 'bootstrap', class = 'table-hover table-condensed', 
                     rownames = FALSE, filter = 'top') %>%
           formatRound(columns = c('p'), digits=3)
```

#### Lower than others

There were `r length(ug[[paste0(cnd, '.dn')]])` such genes

```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
gostplot(rr[[paste0(cnd, '.dn')]])
```

```{r echo=FALSE,out.width='100%'}
rr[[paste0(cnd, '.dn')]]$result %>% 
           dplyr::select(term_name, source, term_size, query_size, 
                         intersect_size = intersection_size, p = p_value) %>% 
           mutate(source = factor(source)) %>%
           datatable(style = 'bootstrap', class = 'table-hover table-condensed', 
                     rownames = FALSE, filter = 'top') %>%
           formatRound(columns = c('p'), digits=3)
```

### Cal27NSD.DKO {.tabset}

```{r echo=FALSE, message=FALSE, warning=FALSE}
cnd <- 'Cal27NSD.DKO'
```

#### Higher than others

There were `r length(ug[[paste0(cnd, '.up')]])` such genes

```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
#gostplot(rr[[paste0(cnd, '.up')]])
```

```{r echo=FALSE,out.width='100%'}
#rr[[paste0(cnd, '.up')]]$result %>% 
       #    dplyr::select(term_name, source, term_size, query_size, 
          #               intersect_size = intersection_size, p = p_value) %>% 
       #    mutate(source = factor(source)) %>%
        ##   datatable(style = 'bootstrap', class = 'table-hover table-condensed', 
        #             rownames = FALSE, filter = 'top') %>%
        #   formatRound(columns = c('p'), digits=3)
```

#### Lower than others

There were `r length(ug[[paste0(cnd, '.dn')]])` such genes

```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
#gostplot(rr[[paste0(cnd, '.dn')]])
```

```{r echo=FALSE,out.width='100%'}
#rr[[paste0(cnd, '.dn')]]$result %>% 
         #  dplyr::select(term_name, source, term_size, query_size, 
         #                intersect_size = intersection_size, p = p_value) %>% 
        #   mutate(source = factor(source)) %>%
        #   datatable(style = 'bootstrap', class = 'table-hover table-condensed', 
        #             rownames = FALSE, filter = 'top') %>%
       #    formatRound(columns = c('p'), digits=3)
```

### Cal27NSD.TKO {.tabset}

```{r echo=FALSE, message=FALSE, warning=FALSE}
cnd <- 'Cal27NSD.TKO'
```

#### Higher than others

There were `r length(ug[[paste0(cnd, '.up')]])` such genes

```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
gostplot(rr[[paste0(cnd, '.up')]])
```

```{r echo=FALSE,out.width='100%'}
rr[[paste0(cnd, '.up')]]$result %>% 
           dplyr::select(term_name, source, term_size, query_size, 
                         intersect_size = intersection_size, p = p_value) %>% 
           mutate(source = factor(source)) %>%
           datatable(style = 'bootstrap', class = 'table-hover table-condensed', 
                     rownames = FALSE, filter = 'top') %>%
           formatRound(columns = c('p'), digits=3)
```

#### Lower than others

There were `r length(ug[[paste0(cnd, '.dn')]])` such genes

```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
gostplot(rr[[paste0(cnd, '.dn')]])
```

```{r echo=FALSE,out.width='100%'}
rr[[paste0(cnd, '.dn')]]$result %>% 
           dplyr::select(term_name, source, term_size, query_size, 
                         intersect_size = intersection_size, p = p_value) %>% 
           mutate(source = factor(source)) %>%
           datatable(style = 'bootstrap', class = 'table-hover table-condensed', 
                     rownames = FALSE, filter = 'top') %>%
           formatRound(columns = c('p'), digits=3)
```

<p>&nbsp;</p>

---

### {-}

# Cal27: K36M-OE vs NSD1/2KO {.tabset .tabset-pills}

<b> For all pairwise comparisons,Cal27OE.K36M is set as reference/baseline. DEGs cutoff padj < 0.05 & |log2FC| > 2. </b>

Many more downregulated genes in NSD1KO compared to K36M-OE whereas NSD2.KO vs K36M-OE results in roughly equal DEGs in either direction. Number of DEGs actually decrease when comparing NSD.DKO to K36M-OE. 

<p>&nbsp;</p>

```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
#Are genes affected by K36M-OE the same ones that change when we perturb NSD1 and/or NSD2?
#k36mOE_v_nsd1ko <- get_dge_df(dds,"group","Cal27NSD1.KO","Cal27OE.K36M")
#k36mOE_v_nsd2ko <- get_dge_df(dds,"group","Cal27NSD2.KO","Cal27OE.K36M")
#k36mOE_v_nsdDko <- get_dge_df(dds,"group","Cal27NSD.DKO","Cal27OE.K36M")
#save/load
load("~/Documents/Q005841_HNSCC/data/k36mOEvNSD1KO_dge.RData")
load("~/Documents/Q005841_HNSCC/data/k36mOEvNSD2KO_dge.RData")
load("~/Documents/Q005841_HNSCC/data/k36mOEvNSD.DKO_dge.RData")
```

## Volcano {.tabset}

### K36M-OE v NSD1KO
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
volc(k36mOE_v_nsd1ko,'log2FoldChange','padj',ylab = "-log10(FDR)",'')
```

### K36M-OE v NSD2KO
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
volc(k36mOE_v_nsd2ko,'log2FoldChange','padj',ylab = "-log10(FDR)",'')
```

### K36M-OE v NSD.DKO
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
volc(k36mOE_v_nsdDko,'log2FoldChange','padj',ylab = "-log10(FDR)",'')
```

## Pathway analysis - upregulated {.tabset}

Comparing NSD1.KO to K36M-OE, cell periphery is an overrepresented pathway, similar to when comparing NSD1.KO to NSD2.KO. 

<p>&nbsp;</p>
Comparing NSD2KO & NSD.DKO to K36M-OE, we find overrepresentation gene sets involved with chromosomal components.

### K36M-OE v NSD1KO
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
#pa_k36mOE_v_nsd1ko <- pathway_analysis(k36mOE_v_nsd1ko,2)
load("~/Documents/Q005841_HNSCC/data/pa_k36mOE_v_nsd1ko.RData")
gostplot(pa_k36mOE_v_nsd1ko$up,capped = FALSE,interactive = TRUE)
reactable(pa_k36mOE_v_nsd1ko$up_reactable)
```

### K36M-OE v NSD2KO
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
#pa_k36mOE_v_nsd2ko <- pathway_analysis(k36mOE_v_nsd2ko,2)
load("~/Documents/Q005841_HNSCC/data/pa_k36mOE_v_nsd2ko.RData")
#save(pa_k36mOE_v_nsd1ko,file="~/Documents/Q005841_HNSCC/data/pa_k36mOE_v_nsd1ko.RData")
gostplot(pa_k36mOE_v_nsd2ko$up,capped = FALSE,interactive = TRUE)
reactable(pa_k36mOE_v_nsd2ko$up_reactable)
```
### K36M-OE v NSD.DKO
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
#pa_k36mOE_v_nsdDko <- pathway_analysis(k36mOE_v_nsdDko,2)
load("~/Documents/Q005841_HNSCC/data/pa_k36mOE_v_nsdDko.RData")
gostplot(pa_k36mOE_v_nsdDko$up,capped = FALSE,interactive = TRUE)
reactable(pa_k36mOE_v_nsdDko$up_reactable)
```

## Pathway analysis - downregulated {.tabset}

Comparing downregulated NSD1.KO, NSD2.KO or NSD.DKO to K36M-OE, similar overrepresented pathways are found as when comparing NSD1.KO to NSD2.KO. Example: cell periphery, multicellular organismal process

### K36M-OE v NSD1KO
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
gostplot(pa_k36mOE_v_nsd1ko$down,capped = FALSE,interactive = TRUE)
reactable(pa_k36mOE_v_nsd1ko$down_reactable)
```

### K36M-OE v NSD2KO
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
gostplot(pa_k36mOE_v_nsd2ko$down,capped = FALSE,interactive = TRUE)
reactable(pa_k36mOE_v_nsd2ko$down_reactable)
```

### K36M-OE v NSD.DKO
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
gostplot(pa_k36mOE_v_nsdDko$down,capped = FALSE,interactive = TRUE)
reactable(pa_k36mOE_v_nsdDko$down_reactable)
```

## GSEA {.tabset}

Upregulated genes seems to have similar enriched pathways.

### K36M-OE v NSD1KO
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
#gsea_k36mOE_v_NSD1KO <- gsea(k36mOE_v_nsd1ko)
#save(gsea_k36mOE_v_NSD1KO,file="~/Documents/Q005841_HNSCC/data/gsea_k36mOE_v_NSD1KO_cal27.RData")
load("~/Documents/Q005841_HNSCC/data/gsea_k36mOE_v_NSD1KO_cal27.RData")
plot(gsea_k36mOE_v_NSD1KO$plot.r)
gsea_k36mOE_v_NSD1KO$gsea_table
```

### K36M-OE v NSD2KO
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
#gsea_k36mOE_v_NSD2KO <- gsea(k36mOE_v_nsd2ko)
#save(gsea_k36mOE_v_NSD2KO,file="~/Documents/Q005841_HNSCC/data/gsea_k36mOE_v_NSD2KO_cal27.RData")
load("~/Documents/Q005841_HNSCC/data/gsea_k36mOE_v_NSD2KO_cal27.RData")
plot(gsea_k36mOE_v_NSD2KO$plot.r)
gsea_k36mOE_v_NSD2KO$gsea_table
```

### K36M-OE v NSD.DKO
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
#gsea_k36mOE_v_NSD.DKO <- gsea(k36mOE_v_nsdDko)
#save(gsea_k36mOE_v_NSD.DKO,file="~/Documents/Q005841_HNSCC/data/gsea_k36mOE_v_NSD.DKO_cal27.RData")
load("~/Documents/Q005841_HNSCC/data/gsea_k36mOE_v_NSD.DKO_cal27.RData")
plot(gsea_k36mOE_v_NSD.DKO$plot.r)
gsea_k36mOE_v_NSD.DKO$gsea_table
```

## GSEA top hit {.tabset}

Note: E2F targets genes that encode proteins involved in DNA replication.

### K36M-OE v NSD1KO
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
plot(gsea_k36mOE_v_NSD1KO$top_hit)
```

### K36M-OE v NSD2KO
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
plot(gsea_k36mOE_v_NSD2KO$top_hit)
```

### K36M-OE v NSD.DKO
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
plot(gsea_k36mOE_v_NSD.DKO$top_hit)
```

## Overlapping DEGS {.tabset}

Overlapping DEGs were obtained from K36M.OE against K36WT, NSD1KO against K36WT, NSD2KO against K36WT & NSD.DKO against K36WT. DEGs were obtained with the following criteria: padj < 0.05 & |log2FC| > 0.5.
<p>&nbsp;</p>
86 DEGs in common between NSD1 & K36M-OE
<p>&nbsp;</p>
94 DEGs in common between NSD2 & K36M-OE
<p>&nbsp;</p>
66 genes in common between NSD.DKO & K36M-OE

### Euler graphs

```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
#need to pa before euler graph
k36m.OE_v_nsd1ko <- overlap_analysis(dds = dds,grouping = "group",cond1 = "Cal27OE.K36M",cond2 = "Cal27NSD1.KO",baseCond = "Cal27OE.K36WT",fc_cutoff = 0.5)
#
k36m.OE_v_nsd2ko <- overlap_analysis(dds = dds,grouping = "group",cond1 = "Cal27OE.K36M",cond2 = "Cal27NSD2.KO",baseCond = "Cal27OE.K36WT",fc_cutoff = 0.5)
#
k36m.OE_v_nsdDko <- overlap_analysis(dds = dds, grouping= "group", cond1 = "Cal27OE.K36M",cond2="Cal27NSD.DKO",baseCond = "Cal27OE.K36WT",fc_cutoff = 0.5)
#
get_euler_graph("K36M.OE","NSD1.KO","NSD2.KO", "NSD.DKO",k36m.OE_v_nsd1ko$x1,k36m.OE_v_nsd1ko$x2,k36m.OE_v_nsd2ko$x2,k36m.OE_v_nsdDko$x2,0.05)
```

### Pathway analysis for overlapping DEGs {.tabset}

Overlapping DEGs also have overrepresentation of pathways related to 'cell periphery' and cellular component.

#### K36M.OE v NSD1KO
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
gostplot(k36m.OE_v_nsd1ko$pa)
reactable(k36m.OE_v_nsd1ko$reactable)
```

#### K36M.OE v NSD2KO
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
gostplot(k36m.OE_v_nsd2ko$pa)
reactable(k36m.OE_v_nsd2ko$reactable)
```

#### K36M.OE v NSD.DKO
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
gostplot(k36m.OE_v_nsdDko$pa)
reactable(k36m.OE_v_nsdDko$reactable)
```

---

<p>&nbsp;</p>

# Detroit: K36M-OE vs NSD1KO {.tabset .tabset-pills}

Fewer DEGs comparing K36M-OE to NSD1KO in Detroit cell line. Downregulated genes are overrepresented in similar pathways as Cal27 cell line. 

```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%',fig.cap="For all pairwise comparisons,Cal27OE.K36M is set as reference/baseline"}
#Are genes affected by K36M-OE the same ones that change when we perturb NSD1 and/or NSD2?
det_k36mOE_v_nsd1ko <- get_dge_df(dds,"group","DetroitNSD1.KO","DetroitOE.K36M")
#save/load
#save(det_k36mOE_v_nsd1ko,file="~/Documents/Q005841_HNSCC/data/det_k36mOE_v_nsd1ko.RData")
load("~/Documents/Q005841_HNSCC/data/det_k36mOE_v_nsd1ko.RData")
```

## Volcano
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
volc(det_k36mOE_v_nsd1ko,'log2FoldChange','padj',ylab = "-log10(FDR)",'')
```

## Pathway analysis - upregulated

Only DEGs with |log2FC|>1 & FDR < 0.05 were inputted. 

```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
#det_pa_k36mOE_v_nsd1ko <- pathway_analysis(det_k36mOE_v_nsd1ko,1)
#may need to adjust fc if few DEGs
#save(det_pa_k36mOE_v_nsd1ko,file="~/Documents/Q005841_HNSCC/data/det_pa_k36mOE_v_nsd1ko.RData")
load("~/Documents/Q005841_HNSCC/data/det_pa_k36mOE_v_nsd1ko.RData")
gostplot(det_pa_k36mOE_v_nsd1ko$up,capped = FALSE,interactive = TRUE)
reactable(det_pa_k36mOE_v_nsd1ko$up_reactable)
```

## Pathway analysis - downregulated

Only DEGs with |log2FC|>1 & FDR < 0.05 were inputted.

```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
gostplot(det_pa_k36mOE_v_nsd1ko$down,capped = FALSE,interactive = TRUE)
reactable(det_pa_k36mOE_v_nsd1ko$down_reactable)
```

## GSEA

GSEA results 

```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
#det_gsea_k36mOE_v_NSD1KO <- gsea(det_k36mOE_v_nsd1ko)
#save(det_gsea_k36mOE_v_NSD1KO,file="~/Documents/Q005841_HNSCC/data/det_gsea_k36mOE_v_NSD1KO_cal27.RData")
load("~/Documents/Q005841_HNSCC/data/det_gsea_k36mOE_v_NSD1KO_cal27.RData")
plot(det_gsea_k36mOE_v_NSD1KO$plot.r)
det_gsea_k36mOE_v_NSD1KO$gsea_table
```

## GSEA top hit
```{r echo = FALSE, warning=FALSE,message=FALSE,out.width='100%'}
plot(det_gsea_k36mOE_v_NSD1KO$top_hit)
```

## Overlapping DEGS {.tabset}

Overlapping DEGs were obtained from K36M.OE against K36WT and NSD1KO against K36WT. DEGs were obtained with the following criteria: padj < 0.05 & |log2FC| > 0.5

Few overlapping DEGs are found.

<p>&nbsp;</p>

Similar 

### Euler graphs
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
#need to do overlap pathway analysis before euler graph
det_k36m.OE_v_nsd1ko <- overlap_analysis(dds = dds,grouping = "group",cond1 = "DetroitOE.K36M",cond2 = "DetroitNSD1.KO",baseCond = "DetroitOE.K36WT",fc_cutoff = 0.5)
#
get_euler_graph_for_two("K36M.OE","NSD1.KO",det_k36m.OE_v_nsd1ko$x1,det_k36m.OE_v_nsd1ko$x2,0.05)
```

### Pathway analysis
```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%'}
gostplot(det_k36m.OE_v_nsd1ko$pa)
reactable(det_k36m.OE_v_nsd1ko$reactable)
```

---

<p>&nbsp;</p>

# NSD2-OE {.tabset .tabset-pills}

## Cal27

In Cal27, we see very few DEGs comparing NSD2-OE to parental, indicating NSD2-OE may indeed return NSD1-KO cells to a NSD1-WT-like state. 

```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%', fig.cap="DEs have |logFC|>2 & FDR < 0.05"}
#Can NSD2-OE return NSD1-KO or NSD1-MT cells to a NSD1-WT-like state?
cal27_nsd2_oe <- get_dge_df(dds = dds,group = "group",sampA = "Cal27NSD1KO.OE.NSD2", ref_sampB = "Cal27Parental")
volc(cal27_nsd2_oe,'log2FoldChange','padj',ylab = "-log10(FDR)",'')
```

## FaDu

FaDu NSD2.OE vs FaDu parental. 

```{r echo=FALSE, message=FALSE, warning=FALSE, out.width='100%',fig.cap="DEs have |logFC|>2 & FDR < 0.05"}
#Can NSD2-OE return NSD1-KO or NSD1-MT cells to a NSD1-WT-like state?
fad_nsd2_oe <- get_dge_df(dds = dds,group = "group",sampA = "FaDuNSD1KO.OE.NSD2", ref_sampB = "FaDuParental")
volc(fad_nsd2_oe,'log2FoldChange','padj',ylab = "-log10(FDR)",'')
```